### PR TITLE
feat(CroReconciler): adds delete strategy to remove snapshots

### PR DESCRIPTION
**Pull request by jjaferson**
# Description

adds delete strategy to remove snapshots to the `cloud-resources-aws-strategies` configmap in the rhmi operator namespace

https://issues.redhat.com/browse/INTLY-7184

## Verification Steps

> Requires a BYOC cluster

- Install the rhmi operator with the `USE_CLUSTER_STORAGE` field set to true 
- Uninstall the operator 
- Verify if the `cloud-resources-aws-strategies` configmap in the rhmi operator namespace has gotten the following fields:
  - `blobstorage.production.deleteStrategy.forceBucketDeletion` set to true
  - `postgres.production.deleteStrategy.SkipFinalSnapshot` set to true 
  - `'redis.production.deleteStrategy.FinalSnapshotIdentifier` set to empty 

- Make sure there are not snapshots created

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by grdryn** [comment]
@jjaferson @briangallagher The e2e test is failing here. Is that related to this PR, or is it some flake that needs to be reported?

File: test-cases/tests/uninstallation/o02-verify-if-cro-operator-removes-remaining-snapshots-for-rds-e.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+```bash
+oc describe configmap cloud-resources-aws-strategies -n redhat-rhmi-operator
+```
+
+4. Ensure that there are no manual or automatic RDS snapshots remaining in the AWS account.
+
+```bash
+aws rds describe-db-snapshots | jq
+```
```
Comment (psturc) [inline test-cases/tests/uninstallation/o02-verify-if-cro-operator-removes-remaining-snapshots-for-rds-e.md:55]: There could be some leftover snapshots from previous clusters present in AWS.

We should either make sure that all other RDS, Elasticache and S3 resources are deleted from AWS prior performing this test case, or make this command more specific, i.e. search for resources based on cluster ID or resourceIds?

**Comment by jjaferson** [comment]
> @jjaferson @briangallagher The e2e test is failing here. Is that related to this PR, or is it some flake that needs to be reported?

I'll check the outcome of the next e2e test, but it should not be failing due to the changes in this PR.

File: test-cases/tests/uninstallation/o02-verify-if-cro-operator-removes-remaining-snapshots-for-rds-e.md (hunk 1 above)
Comment (jjaferson) [reply test-cases/tests/uninstallation/o02-verify-if-cro-operator-removes-remaining-snapshots-for-rds-e.md:55]: I'll add a point in the prerequisites that all the leftover snapshots from previous installation should be removed before performing the test case.

File: test-cases/tests/uninstallation/o02-verify-if-cro-operator-removes-remaining-snapshots-for-rds-e.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+5. Ensure that there are no manual or automatic Elasticache snapshots remaining in the AWS account.
+
+```bash
+aws elasticache describe-db-snapshots | jq
+```
+
+6. Ensure that there are no S3 buckets for the cluster remaining in the AWS account.
+
+```bash
+aws s3 ls
```
Comment (psturc) [inline test-cases/tests/uninstallation/o02-verify-if-cro-operator-removes-remaining-snapshots-for-rds-e.md:66]: here we can do
```suggestion
aws_s3_buckets=$(aws s3 ls)
for b in $(oc get blobstorages -n redhat-rhmi-operator -o json | jq -r .items[].metadata.annotations.resourceIdentifier); do echo $aws_s3_buckets | grep $b || echo bucket $b not found ; done
```
and then mention that we should get `bucket not found` in the output (twice)
wdyt?

**Comment by jjaferson** [comment]
> I'll check the outcome of the next e2e test, but it should not be failing due to the changes in this PR.

e2e test failed again with the verification of smtp config for 3scale, it's probably a flaky test and I created a jira for this https://issues.redhat.com/browse/INTLY-9596

**Comment by jjaferson** [comment]
failed verification of smtp config for 3scale again
https://issues.redhat.com/browse/INTLY-9596


/test e2e

**Pull request merged**